openwisp_radius/saml/views.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from urllib.parse import parse_qs, urlparse
+
+import swapper
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured, ObjectDoesNotExist
+from django.shortcuts import get_object_or_404, render
+from djangosaml2.views import (
+    AssertionConsumerServiceView as BaseAssertionConsumerServiceView,
+)
+from djangosaml2.views import LoginView as BaseLoginView
+from djangosaml2.views import LogoutInitView, LogoutView, MetadataView  # noqa
+from rest_framework.authtoken.models import Token
+
+from .. import settings as app_settings
+from ..api.views import RadiusTokenMixin
+from ..utils import load_model
+from .utils import get_url_or_path
+
+logger = logging.getLogger(__name__)
+
+Organization = swapper.load_model('openwisp_users', 'Organization')
+RadiusToken = load_model('RadiusToken')
+RegisteredUser = load_model('RegisteredUser')
+OrganizationUser = swapper.load_model('openwisp_users', 'OrganizationUser')
+
+
+class OrganizationSamlMixin(object):
+    def get_org_slug_from_relay_state(self):
+        try:
+            parsed_url = urlparse(
+                self.request.POST.get(
+                    'RelayState', self.request.GET.get('RelayState', None),
+                )
+            )
+
+            org_slug = parse_qs(parsed_url.query)['org'][0]
+        except (KeyError, IndexError):
+            raise ImproperlyConfigured('Organization slug not provided')
```

Comment:
let's not use `ImproperlyConfigure`, it's not suited for these kind of exceptions, here it's better to use a plain `ValueError`